src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -96,49 +95,38 @@ export default class LimitedRestorationPanel extends Vue {
   }
 
   /**
-   * Called when months input has changed.
+   * Called when radio value or input value have changed.
    */
-  async onMonthsInput (input: string): Promise<void> {
-    // ignore non-inputs
-    if (input !== null) {
-      // wait for component updates
+  @Watch('radioValue')
+  @Watch('inputValue')
+  private async onValueChanged (): Promise<void> {
+    if (this.radioValue === 'customMonths') {
+      // wait for component updates then validate the input field
       await this.$nextTick()
-      // validate the input
       const valid = this.$refs.monthsRef.validate()
-      // emit validity
+
+      // emit months and validity
+      this.emitMonths(valid ? +this.inputValue : null) // emit null if invalid
```

Comment:
Nice.

Reply:
Thanks. Yes, this way this component always emits something reasonable without events depending on other events.

Reply:
The other reason I did this is that it allows the parent component to do something different if invalid. In this case (next PR), the parent will clear any existing expiry date.